mailbox/plugin/quota-mailing/src/main/java/org/apache/james/eventsourcing/CommandDispatcher.java

```diff
@@ -58,12 +61,14 @@ public void dispatch(Command c) {
             .orElseThrow(RuntimeException::new);
     }
 
-    public boolean tryDispatch(Command c) {
+    private boolean tryDispatch(Command c) {
         try {
-            Optional<List<? extends Event>> events = Optional.ofNullable(
-                handlers.get(c.getClass()))
-                    .map(f -> f.apply(c));
-            eventBus.publish(events.orElse(ImmutableList.of()));
+            List<? extends Event> events =
+                Optional.ofNullable(handlers.get(c.getClass()))
+                    .map(f -> f.handle(c))
+                    .orElse(ImmutableList.of());
```

Comment:
in fact we probably need a log in this case because we should not have unhandled Command, don't you think ?

Reply:
Very true.